### PR TITLE
Add custom rx operators ip_delayElements and ip_timeout

### PR DESCRIPTION
**Pull request by Tom-Le**
Add a couple of custom operators to the Rx subspec, plus unit tests.

- `ip_delayElements` to delay elements matching a predicate.
- `ip_timeout` to emit an element whenever a timeout occurs (i.e. whenever observable emits an element, then emits nothing for some time).

I'd love to hear if these are useful additions to swift wisdom, and I could definitely use some help with naming these operators and proofreading their doc comments.

### CODE REVIEW

**Comment by vinnyt** [comment]
I like it. +1

File: SwiftWisdomTests/Rx/Rx+DelayElementsTests.swift [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+//
+
+import XCTest
+import RxSwift
+import RxTest
+@testable import SwiftWisdom
+
+final class RxDelayElementsTests: XCTestCase {
+
+    func testDelayingElements() {
```
Comment (Benuuu) [inline SwiftWisdomTests/Rx/Rx+DelayElementsTests.swift]: Could you also add a test of the same sequence with the predicate ($0 % 2 == 0)

It should expect `[1,3,0,2,4]`

**Review by Benuuu — APPROVED** [review]

File: SwiftWisdom/Rx/Rx+Timeout.swift [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+     Note that unlike `timeout(_,scheduler:)` and `timeout(_,other:scheduler:)`, the returned observable does not
+     terminate after a timeout.
+
+     - parameter dueTime: Maximum duration between emissions before a timeout occurs.
+     - parameter element: Element to be emitted when a timeout occurs.
+     - parameter scehduler: Scheduler to run the timeout timer on.
+     
+     - returns: New observable sequence as described.
+     */
+    public func ip_timeout(
```
Comment (bobgilmore) [inline SwiftWisdom/Rx/Rx+Timeout.swift]: I wish that the name (or argument list) made it clear that this repeats, unlike the other `timeout` methods. Maybe something like `repeatingTimeouts`. The plural, especially, would makke that clear.

File: SwiftWisdomTests/Rx/Rx+DelayElementsTests.swift (hunk 1 above)
Comment (Tom-Le) [reply SwiftWisdomTests/Rx/Rx+DelayElementsTests.swift]: Added another unit test as requested!

File: SwiftWisdom/Rx/Rx+Timeout.swift (hunk 2 above)
Comment (Tom-Le) [reply SwiftWisdom/Rx/Rx+Timeout.swift]: I renamed it to `ip_repeatingTimeouts(interval:element:scheduler:)`. Does that make it clearer?

**Review by paulrolfe — APPROVED** [review]

File: SwiftWisdom/Rx/Rx+RepeatingTimeouts.swift [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+     Note that unlike `timeout(_,scheduler:)` and `timeout(_,other:scheduler:)`, the returned observable does not
+     terminate after a timeout.
+
+     - parameter dueTime: Maximum duration between emissions before a timeout occurs.
+     - parameter element: Element to be emitted when a timeout occurs.
+     - parameter scehduler: Scheduler to run the timeout timer on.
+     
+     - returns: New observable sequence as described.
+     */
+    public func ip_repeatingTimeouts(
```
Comment (bobgilmore) [inline SwiftWisdom/Rx/Rx+RepeatingTimeouts.swift:26]: I like the name change. Renaming the external name of interval to repeatingInterval probably would have worked, too.

**Pull request merged**